crates/dht/src/dht.rs

```diff
@@ -379,15 +389,14 @@ impl DhtApi for Dht {
                             let (op_ids, ub) = self
                                 .store
                                 .retrieve_op_hashes_in_time_slice(
-                                    arc,
-                                    start,
-                                    end,
-                                    Some(max_op_data_bytes - used_bytes),
+                                    arc, start, end,
                                 )
                                 .await?;
-                            out.extend(op_ids);
 
-                            used_bytes += ub;
+                            if (used_bytes + ub) as i32 <= max_op_data_bytes {
+                                out.extend(op_ids);
+                                used_bytes += ub;
+                            }
                         }
                     }
```

Comment:
This can break early if max ops have been reached. There wouldn't be traces then.
```suggestion
                    } else {
                        break 'outer;
                    }
```

Reply:
I was thinking of letting the loop proceed to look for a slice that's smaller and might fit into the limit. But maybe that's a lot of wasteful database queries. I think it'd be reasonable to break if the limit is reached or exceeded but probably not if the current slice is too big. I'll take a look at this